src/au3audio/internal/audioengine.cpp

```diff
@@ -3,18 +3,46 @@
 #include "libraries/lib-audio-io/AudioIO.h"
 #include "libraries/lib-audio-io/ProjectAudioIO.h"
 
-#include "libraries/lib-track/Track.h"
+#include "libraries/lib-module-manager/PluginManager.h"
+#include "libraries/lib-wave-track/WaveTrack.h"
 #include "libraries/lib-time-frequency-selection/ViewInfo.h"
+#include "libraries/lib-effects/EffectManager.h"
+#include "libraries/lib-realtime-effects/RealtimeEffectState.h"
+#include "libraries/lib-realtime-effects/RealtimeEffectList.h"
 
 #include "au3wrap/au3types.h"
+#include "au3wrap/internal/domaccessor.h"
+#include "au3wrap/internal/domconverter.h"
 
 #include "defaultplaybackpolicy.h"
 #include "au3audioiolistener.h"
+#include "project/iaudacityproject.h"
 
 #include "realfn.h"
 
 using namespace au::audio;
 
+namespace {
+std::string getEffectName(const std::string& effectId)
+{
+    const auto desc = PluginManager::Get().GetPlugin(effectId);
+    IF_ASSERT_FAILED(desc) {
+        return "";
+    }
+    return desc->GetSymbol().Msgid().Translation().ToStdString();
+}
```

Comment:
It is better to add the required method to the `IEffectsProvider` and use it through injection